site/src/components/AuthProvider/AuthProvider.tsx

```diff
@@ -1,36 +1,127 @@
-import { useActor, useInterpret } from "@xstate/react";
+import { checkAuthorization } from "api/queries/authCheck";
+import {
+  authMethods,
+  hasFirstUser,
+  login,
+  logout,
+  me,
+  updateProfile as updateProfileOptions,
+} from "api/queries/users";
+import {
+  AuthMethods,
+  UpdateUserProfileRequest,
+  User,
+} from "api/typesGenerated";
 import { createContext, FC, PropsWithChildren, useContext } from "react";
-import { authMachine } from "xServices/auth/authXService";
-import { ActorRefFrom } from "xstate";
+import { useMutation, useQuery, useQueryClient } from "react-query";
+import { permissionsToCheck, Permissions } from "./permissions";
+import { displaySuccess } from "components/GlobalSnackbar/utils";
+import { FullScreenLoader } from "components/Loader/FullScreenLoader";
+import { isApiError } from "api/errors";
 
-interface AuthContextValue {
-  authService: ActorRefFrom<typeof authMachine>;
-}
+type AuthContextValue = {
+  isSignedOut: boolean;
+  isSigningOut: boolean;
+  isConfiguringTheFirstUser: boolean;
+  isSignedIn: boolean;
+  isSigningIn: boolean;
+  isUpdatingProfile: boolean;
+  user: User | undefined;
+  permissions: Permissions | undefined;
+  authMethods: AuthMethods | undefined;
+  signInError: unknown;
+  updateProfileError: unknown;
+  signOut: () => void;
+  signIn: (email: string, password: string) => Promise<void>;
+  updateProfile: (data: UpdateUserProfileRequest) => void;
+};
 
 const AuthContext = createContext<AuthContextValue | undefined>(undefined);
 
 export const AuthProvider: FC<PropsWithChildren> = ({ children }) => {
-  const authService = useInterpret(authMachine);
+  const meOptions = me();
+  const userQuery = useQuery(meOptions);
+  const authMethodsQuery = useQuery(authMethods());
+  const hasFirstUserQuery = useQuery(hasFirstUser());
+  const permissionsQuery = useQuery({
+    ...checkAuthorization({ checks: permissionsToCheck }),
+    enabled: userQuery.data !== undefined,
+  });
+
+  const queryClient = useQueryClient();
+  const loginMutation = useMutation(
+    login({ checks: permissionsToCheck }, queryClient),
+  );
+  const logoutMutation = useMutation(logout(queryClient));
+  const updateProfileMutation = useMutation({
+    ...updateProfileOptions(),
+    onSuccess: (user) => {
+      queryClient.setQueryData(meOptions.queryKey, user);
+      displaySuccess("Updated settings.");
+    },
+  });
+
+  const isSignedOut =
+    userQuery.isError &&
+    isApiError(userQuery.error) &&
+    userQuery.error.response.status === 401;
+  const isSigningOut = logoutMutation.isLoading;
+  const isLoading =
+    authMethodsQuery.isLoading ||
+    userQuery.isLoading ||
+    hasFirstUserQuery.isLoading ||
+    (userQuery.isSuccess && permissionsQuery.isLoading);
```

Comment:
Yeah, that's fair. This might be partly misunderstanding the API on my part – I need to review what happens if a query succeeds on a fetch, but errors out on a refetch